### ISSUE TITLE
BatchNormalization error in low version?

### ISSUE THREAD

**Issue by triumph-wangyuyang** [tensorflow/tensorflow#58711]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Others

### Source

source

### Tensorflow Version

TF 2.4, TF 2.10

### Custom Code

Yes

### OS Platform and Distribution

windows 11

### Python version

3.8.15

### CUDA/cuDNN version

CUDA: 11.3    cuDNN 8.6.0

### GPU model and memory

RTX3060

### Current Behaviour?

```shell
Why does tf.keras.layers.BatchNormalization error occur in the low version (tensorflow2.4), but not in the high version (tensorflow2.13)?
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
    m = tf.keras.layers.BatchNormalization(axis=-1, epsilon=0.001, momentum=0.999)
    input = tf.saturate_cast(tf.random.uniform([1, 3, 1, 672], minval=0, maxval=256, dtype=tf.int64), dtype=tf.uint8)
    out = m(input)
    print(out)
```

### Relevant log output

_No response_</details>

**Comment by tilakrayal** [comment]
@triumph-wangyuyang,
In tensorflow v2.4, the mentioned exception was there but whereas in the latest tf version, it was removed and was able to execute without any issue/error. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/6e7eef8725c5607410c22cd6fc9837ae/untitled779.ipynb). Thank you!

**Comment by triumph-wangyuyang** [comment]
> 

Thanks for your reply, I have tested it in the new version of tf.
